### DISCUSSION TITLE
Virtual Output issue

### DISCUSSION THREAD

**Discussion by razvanbalint** [orchetect/swift-midi#161] [Q&A] [answered]
I'm trying to create an app that outputs MIDI commands to my macbook. 
I've read the documentation for this library and also downloaded and tested Virtual Output example project (which is awesome btw, thanks for your effort). 
I've configured MIDI Studio (create session and connect my iPhone to it) and I've downloaded a software called MidiView to inspect traffic. 
But somehow, no traffic appears there. What am I missing? 
There is no error thrown when starting the manager or creating the output port. 

However, I've also downloaded this app from AppStore called MIDI Wrench (something similar to what I'm trying to achieve). Using the same setup mentioned above, MidiView is able to intercept traffic from it. 

Do you have any idea why Virtual Output might not work properly for me? I cannot find any issue with the code or my setup.

**Comment by orchetect** [comment, marked as answer]
A Virtual Output will only work with other software on the iOS device.

Need more details. How exactly are you connecting?

MIDI over Bluetooth, MIDI over Network, and MIDI over USB are all separate things and not directly related. Check out the docs for each of these [here](https://orchetect.github.io/MIDIKit/documentation/midikit/midikit-getting-started/#3-Additional-Connectivity-Bluetooth-Network-USB). There's also example projects to help out.

**Reply by orchetect** [reply]
I think by "create session" you mean network MIDI.

Be aware that there is a macOS bug where these live routings do not work:
![midi-network-live-routings](https://user-images.githubusercontent.com/11605557/215188304-174a6b3f-401b-4cfc-a8d8-332aa5e535e8.png)

Also, in a Network MIDI session, it will auto-create MIDI endpoints on the Mac and the iOS device related to the session name. You would need to use `addOutputConnection` to connect to them.

**Reply by orchetect** [reply]
I've updated the online docs to include this information and a more thorough guide:

https://orchetect.github.io/MIDIKit/documentation/midikit/midikitio-midi-over-network

**Reply by razvanbalint** [reply]
Thanks for your response. I was trying to create a new output instead of connecting to the existing one.
